src/main/java/bio/terra/workspace/app/configuration/spring/TraceInterceptorConfig.java

```diff
@@ -0,0 +1,96 @@
+package bio.terra.workspace.app.configuration.spring;
+
+import static org.apache.commons.lang3.ObjectUtils.getFirstNonNull;
+
+import bio.terra.workspace.app.configuration.external.TracingConfiguration;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.samplers.Samplers;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.hashids.Hashids;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+@ConditionalOnProperty("workspace.tracing.enabled")
+public class TraceInterceptorConfig implements WebMvcConfigurer {
+
+  public static final String MDC_REQUEST_ID_HEADER = "X-Request-ID";
+  public static final String MDC_REQUEST_ID_KEY = "requestId";
+  public static final String MDC_CORRELATION_ID_HEADER = "X-Correlation-ID";
+
+  private final Hashids hashids = new Hashids("requestIdSalt", 8);
+
+  @Autowired
+  public TraceInterceptorConfig(TracingConfiguration tracingConfiguration) {
+
+    TraceConfig globalTraceConfig = Tracing.getTraceConfig();
+    globalTraceConfig.updateActiveTraceParams(
+        globalTraceConfig.getActiveTraceParams().toBuilder()
+            .setSampler(Samplers.probabilitySampler(tracingConfiguration.getProbability()))
+            .build());
```

Comment:
move this into StackdriverTraceExporterConfig - it's the general setup of tracing as well.

Reply:
I'd been thinking of this as not being stackdriver specific, are we expecting different rules for different tracers?

Reply:
@wchamber is that 👍 makes sense, or 👍 it is stackdriver-specific and should be moved?

Reply:
thumbs up that makes sense. Leave where is. Sorry for the ambiguity.